woocommerce/woocommerce-ios
# Hub Menu: handle the remaining actions, like Switch Store, Settings and WooCommerce Admin

Part of #5509 

### Description
In this PR I handled all the remaining actions of the Hub Menu, implementing the possibility to
- Switch Store
- Open Settings
- Open the WooCommerce Admin login page

## Main Changes
- Implemented the `SettingsView` that conforms to `UIViewControllerRepresentable` for presenting the `SettingsViewController` from the HubMenu.
- Now the `HubMenu` accept an optional `UINavigationController` that will be used from the `StorePickerCoordinator` for presenting the `StorePickerViewController` based on the configuration choosed.
- Implemented `observeSiteForUIUpdates`, for observing the site for any UI updates that depend on it like the store name, siteID, etc…

### Testing instructions
1. Launch the app in debug mode (because of the feature flag).
2. You should be able to see the new tab "Menu".
3. Tap on `Switch Store` button in the navbar. You should be able to change the store and all the info in the Hub Menu should be updated.
4. Tap the `Settings` button, you should be able to use all the sub-functionalities like before.
5. Tap the `WooCommerce Admin` menu. You should be able to view the login page of WP-Admin or WPCom.


---
